workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/RootScreen.kt

```diff
@@ -0,0 +1,49 @@
+package com.squareup.workflow1.ui.container
+
+import com.squareup.workflow1.ui.Compatible
+import com.squareup.workflow1.ui.Screen
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewRegistry
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.updateFrom
+
+/**
+ * Pairs a [screen] rendering with a [viewEnvironment] to support its display.
+ * Typically the rendering type (`RenderingT`) of the root of a UI workflow,
+ * but can be used at any point to modify the [ViewEnvironment] received from
+ * a parent view.
+ */
```

Comment:
* Introduces the Screen marker interface, similar to the one in workflow-swift
* Introduces the RootScreen rendering type as the preferred way to manage the ViewEnvironment, and makes WorkflowLayout use it

> What is RootScreen

I guess I'm trying more to understand what does "Root" mean in the context of Workflows here. In my mind Root almost means singular, but this seems more generic.

Reply:
Makes sense as a utility. I think that we need a new dominant metaphor for naming this as the 'tree' metaphor is lacking here when talking about transforming the ViewEnvironment.